module/zfs/zio.c

```diff
@@ -2841,6 +2848,16 @@ zio_gang_issue(zio_t *zio)
 	return (zio);
 }
 
+static void
+zio_gang_inherit_allocator(zio_t *pio, zio_t *cio)
+{
+	if (ZIO_HAS_ALLOCATOR(pio)) {
+		ASSERT(!ZIO_HAS_ALLOCATOR(cio));
```

Comment:
Do we need those checks? Can't we just assign always?

Reply:
removed, thanks